internal/converter/internal/test_common/testing.go

```diff
@@ -214,14 +217,42 @@ func attemptLoadingAlloyConfig(t *testing.T, bb []byte) {
 			clusterService,
 			labelstore.New(nil, prometheus.DefaultRegisterer),
 			remotecfgService,
+			livedebugging.New(),
 		},
 		EnableCommunityComps: true,
 	})
 	err = f.LoadSource(cfg, nil, "")
 
+	// This is a bit of an ugly workaround but the spanmetrics connector is starting a go routine in its start function.
```

Comment:
I guess this is referring to the start function of OTel Collector components? Do we call this when we call  Alloy's`New()` function for a component? I think those `New()` functions aren't meant to have side effects such as opening files and starting goroutines. They should only do this once Run is called.

Reply:
Thank you for pointing this out... I added a [comment](https://github.com/grafana/alloy/pull/2262#discussion_r1886472989) to the other PR. `New` is not supposed to have such side effects.